daemon/api.go

```diff
@@ -609,6 +610,13 @@ func searchStore(c *Command, r *http.Request, user *auth.UserState) Response {
 		Private: private,
 		Prefix:  prefix,
 	}, user)
+	if e, ok := err.(net.Error); ok && e.Timeout() {
```

Comment:
Would it make sense to put the if under `default` case in the `switch` just below?